timon/probes.py

```diff
@@ -333,3 +362,79 @@ def probe_action(self):
 
 class DiskFreeProbe(Probe):
     pass
+
+
+class HttpJsonIntervalProbe(HttpProbe):
+    """
+    probe checking if a value is:
+        - between 2 values (example: "key1.key2:[0, 20]")
+        - greater than a value (example: "key1.key2.key3>60")
+        - lesser than a value (example: "key<20")
+        - equal to a value (example: "key1.key2:200")
+    """
+    script_module = "timon.scripts.http_json"
+
+    def __init__(self, **kwargs):
+        self.ok_rule = kwargs.pop('ok_rule', None)
+        self.warning_rule = kwargs.pop('warning_rule', None)
+        super().__init__(**kwargs)
+
+    def match_rule(self, rslt, rule):
+        rule_types = {
+            "equal_rule": re.compile(r"^(.*):([a-zA-Z0-9]+)$"),
+            "greater_rule": re.compile(r"^(.*)>(\d+)$"),
+            "lesser_rule": re.compile(r"^(.*)<(\d+)$"),
+            "interval_rule": re.compile(r"^(.*):\[(\d+),\s*(\d+)\]$"),
+            }
+
+        def define_rule(rule):
+            for rule_type, rule_rex in rule_types.items():
+                match = rule_rex.match(rule)
+                if match:
+                    return rule_type, match
+            return None
```

Comment:
it is possible but you could do return None, None
This means the caller requires less code as you can just write

```
rule_type, match = match_rule(...)
if rule_type is not None:
    ...
```